File: SmartBooks.Domains/Entities/Bank.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace SmartBooks.Domains.Entities
{
    public class Bank : BaseEntity
    {
        public Bank()
        {
            BankBranches = new HashSet<BankBranch>();
        }
        public int CountryId { get; set; }
        [StringLength(50)]
        public string BankCode { get; set; }
        [StringLength(150)]
        public string BankName { get; set; }

        public Country Country { get; set; }
        public ICollection<BankBranch> BankBranches { get; set; }

    }
}

File: SmartBooksFrontend/src/app/models/bank.ts
import { NumberValueAccessor } from '@angular/forms';

export interface Bank {
  id: number;
  countryId: number;
  bankCode: string;
  bankName: string;
  bankBranches: Array<BankBranch>;
}

export interface BankBranch {
  id: number;
  bankId: number;
  branchCode: string;
  branchName: string;
  swiftCode: string;
}

